src/ui/draggablepixmapitem.cpp

```diff
@@ -58,10 +58,21 @@ void DraggablePixmapItem::bindToUserData(QComboBox *combo, QString key) {
 
 void DraggablePixmapItem::mousePressEvent(QGraphicsSceneMouseEvent *mouse) {
     active = true;
-    last_x = static_cast<int>(mouse->pos().x() + this->pos().x()) / 16;
-    last_y = static_cast<int>(mouse->pos().y() + this->pos().y()) / 16;
+    QPoint pos = Metatile::coordFromPixmapCoord(mouse->scenePos());
+    last_x = pos.x();
+    last_y = pos.y();
     this->editor->selectMapEvent(this, mouse->modifiers() & Qt::ControlModifier);
     this->editor->selectingEvent = true;
+    if (this->editor->obj_edit_mode == "select") {
```

Comment:
all of the logic inside this `if` block really belongs in the `MapRuler` class.

Reply:
True, I thought about doing that initially, but I was trying to match the way other `if` blocks in this function did things by parsing inputs in this function. I'll change this.

Reply:
Wait, my bad, I thought this was the `if` block in `editor.cpp`. In this case I was just matching that `if` block.

Reply:
I'm not sure which one you're talking about.  All I know is the block I commented on should probably be moved into `MapRuler`

Reply:
I agree. And the block I was talking about is in `Editor::mouseEvent_map()` which almost exactly matches this `if` block. So if I change this one I'll probably change that one as well.